### PR TITLE
Simplify flow diagrams

### PR DESCRIPTION
**Pull request by acoburn**
This simplifies the flow diagrams, removing those actors that are out of scope for this specification and placing the common portions of the flow into its own diagram. And given that `source` and `target` are not defined until later in the document, this uses more generally understood terminology to describe the categories of interaction.

PR towards https://github.com/solid/notifications/issues/77

### CODE REVIEW

File: protocol.html [hunk 1]
```diff
@@ -384,19 +384,29 @@ <h3 property="schema:name">Overview</h3>
...
+                    <figcaption property="schema:name">Solid Subscription Flow for Connected Clients</figcaption>
                   </figure>
 
+                  <p>Subscription types in which a client establishes a subscription and does not maintain a persistent connection to a Notifications Source are illustrated by the following diagram.</p>
+
                   <figure id="solid-notifications-flow-target" rel="schema:hasPart" resource="#solid-notifications-flow-target">
                     <img src="notifications-flow-target.mmd.svg" rel="schema:image" width="800" />
 
-                    <figcaption property="schema:name">Solid Notifications Flow (target)</figcaption>
+                    <figcaption property="schema:name">Solid Notifications Flow for Offline Clients</figcaption>
```
Comment (elf-pavlik) [inline protocol.html:409]: I don't think *Offline Clients* makes sense here.
Actually, client-server terminology might be confusing since the Subscriber acts as the server when a notification gets delivered, and the notifications publisher will act as a client.

Actually, the Subscriber needs to maintain their `target` online to be able to receive incoming requests with notifications.

**Review by elf-pavlik — CHANGES_REQUESTED** [review]

File: protocol.html (hunk 1 above)
Comment (elf-pavlik) [reply protocol.html:409]:
```suggestion
                    <figcaption property="schema:name">Solid Notifications Flow, where the publisher connects to the notification target (from the <a href="#subscription-request">subscription request</a>)</figcaption>                    
```

again `target` chan be changed in dedicated PR

**Comment by elf-pavlik** [comment]
I think separating
* Subscription Flow
* Notifications Flow

into separate diagrams is very helpful!

**Comment by elf-pavlik** [comment]
Another option: 

```mermaid
sequenceDiagram
  autonumber
  participant Notifications Target
  participant Subscriber
  participant Storage Description Resource
  participant Subscription Resource

  Subscriber ->> Storage Description Resource: Discovery
  Storage Description Resource ->> Subscriber: Storage Description Resource
  Subscriber ->> Subscription Resource: Request subscription (with access token)
  Subscription Resource ->> Subscriber: Subscription response (with subscription information)

  alt pick-up
    Subscriber ->> Notifications Source: Establish connection
    Notifications Source -->> Subscriber: Stream notifications
  else
    loop for each notification
      Notifications Source ->> Notifications Target: Deliver notification
      Notifications Target ->> Notifications Source: 200 OK
    end
  end

  Subscriber ->> Subscription Resource: Unsubscribe (delete subscription)
  Subscription Resource ->> Subscriber: 204 No Content
```

**Comment by elf-pavlik** [comment]
I think separate

* Subscription Flow (with Unsubscribe)
* Notification Flow

Would be the cleanest approach.

```mermaid
sequenceDiagram
  autonumber
  participant Subscriber
  participant Storage Description Resource
  participant Subscription Resource

  Subscriber ->> Storage Description Resource: Discovery
  Storage Description Resource ->> Subscriber: Storage Description Resource
  Subscriber ->> Subscription Resource: Subscription Request (with access token)
  Subscription Resource ->> Subscriber: Subscription Response (with subscription information)

  Note over Subscriber, Subscription Resource: Notifications Flow

  Subscriber ->> Subscription Resource: Unsubscribe (delete subscription)
  Subscription Resource ->> Subscriber: 204 No Content
```

```mermaid
sequenceDiagram
  autonumber
  participant Subscriber
  participant Notifications Source

  Subscriber ->> Notifications Source: Establish connection
  Notifications Source -->> Subscriber: Stream notifications
```

```mermaid
sequenceDiagram
  autonumber
  participant Notifications Target
  participant Notifications Publisher

  loop for each notification
    Notifications Publisher ->> Notifications Target: Deliver notification
    Notifications Target ->> Notifications Publisher: 200 OK
  end
```

In the implementation I'm working on, which implements the subscriber part, I have separate background job queues for subscribing and handing incoming notifications. There are separate workers handling subscriptions and notifications which could eventually run as independent microservices.

**Comment by elf-pavlik** [comment]
I think we should add an inline issue referencing #106 and once it gets resolved update all affected naming.

**Comment by acoburn** [comment]
@elf-pavlik I have updated the diagrams and terminology to be in line with what was discussed in the 10 Nov meeting

**Review by elf-pavlik — APPROVED** [review]
LGTM

File: notifications-flow-target.mmd [hunk 2]
```diff
@@ -1,20 +1,9 @@
 sequenceDiagram
   autonumber
-  participant Subscriber
+  participant Notifications Publisher
```
Comment (csarven) [inline notifications-flow-target.mmd:3]:
```suggestion
  participant Notifications Sender
```

File: notifications-flow-target.mmd [hunk 3]
```diff
@@ -1,20 +1,9 @@
...
-  Subscriber ->> Storage Metadata Resource: Discovery
-  Storage Metadata Resource ->> Subscriber: Storage Metadata Resource
-  Subscriber ->> Subscriptions Container: Request subscription (with access token)
-  Subscriptions Container ->> Authorization Server: Verify authorization
-  Authorization Server ->> Subscriptions Container: capabilities
-  Subscriptions Container ->> Subscriber: Subscription response (with webid & unsubscribe)
   loop for each notification
-    Notifications Source ->> Notifications Target: Deliver notification
+    Notifications Publisher ->> Notifications Target: Deliver notification
+    Notifications Target ->> Notifications Publisher: 200 OK
```
Comment (csarven) [inline notifications-flow-target.mmd:8]:
```suggestion
    Notifications Sender ->> Notifications Target: Deliver notification
    Notifications Target ->> Notifications Sender: 200 OK
```

File: protocol.html [hunk 4]
```diff
@@ -385,19 +385,29 @@ <h3 property="schema:name">Overview</h3>
...
+                    <figcaption property="schema:name">Solid Subscription Flow, where the subscriber connects to the notification source</figcaption>
                   </figure>
 
+                  <p>Subscription types in which a client establishes a subscription and does not maintain a persistent connection to a Notifications Source are illustrated by the following diagram.</p>
+
                   <figure id="solid-notifications-flow-target" rel="schema:hasPart" resource="#solid-notifications-flow-target">
                     <img src="notifications-flow-target.mmd.svg" rel="schema:image" width="800" />
 
-                    <figcaption property="schema:name">Solid Notifications Flow (target)</figcaption>
+                    <figcaption property="schema:name">Solid Notifications Flow, where the publisher connects to the notification target</figcaption>
```
Comment (csarven) [inline protocol.html:410]:
```suggestion
                    <figcaption property="schema:name">Solid Notifications Flow, where the sender delivers to the notification target</figcaption>
```

**Review by csarven — APPROVED** [review]
I understand the intention is about creating and distributing the notification and "publisher" suffices. Just want to note that publishing may be generally associated with publishing (and serving) a document on a Web site.

"Sender" (or "producer") of content may also work. "Sender" (alongside "recipient) is also terminology that's common/used in HTTP and LDN (FWIW).

Approving so that we move the work forward and not get stuck on this. We do however need to revisit all terminology, especially in the context of JSON-LD and the Notify vocabulary: https://github.com/solid/vocab/pull/62 (and all the issues touching it / touched by it).

**Comment by elf-pavlik** [comment]
> Just want to note that publishing may be generally associated with publishing (and serving) a document on a Web site.

Besides publishing the notifications, the *publisher* also publishes the *topic* to which subscriber wants to subscribe to.

I'm open to further iterations on the terminology and fully agree that this PR as it stands makes a good step forward.

**Comment by csarven** [comment]
>Besides publishing the notifications, the publisher also publishes the topic to which subscriber wants to subscribe to.

I don't quite follow the context of that sentence. What's the significance of "topic" besides it being a unit of information in a notification? Is the "publisher" intended to manage the "topic" resource that's of interest to the subscriber? As I understand it, the "publisher" is a role of creating/producing/sending/delivering/distributing a notification, and that the publication/having a representation of the topic resource is done by some resource server.

File: notifications-flow-target.mmd [hunk 5]
```diff
@@ -1,20 +1,9 @@
...
-  Storage Metadata Resource ->> Subscriber: Storage Metadata Resource
-  Subscriber ->> Subscriptions Container: Request subscription (with access token)
-  Subscriptions Container ->> Authorization Server: Verify authorization
-  Authorization Server ->> Subscriptions Container: capabilities
-  Subscriptions Container ->> Subscriber: Subscription response (with webid & unsubscribe)
   loop for each notification
-    Notifications Source ->> Notifications Target: Deliver notification
+    Notifications Sender ->> Notifications Target: Deliver notification
+    Notifications Target ->> Notifications Sender: 200 OK
   end
```
Comment (csarven) [inline notifications-flow-target.mmd:9]: If `loop` is removed:

```suggestion
```

**Review by csarven — CHANGES_REQUESTED** [review]
Requesting changes mainly for the mention of removing "access token".

File: notifications-flow-target.mmd [hunk 6]
```diff
@@ -1,20 +1,9 @@
...
-  participant Authorization Server
-  participant Notifications Source
 
-  Subscriber ->> Storage Metadata Resource: Discovery
-  Storage Metadata Resource ->> Subscriber: Storage Metadata Resource
-  Subscriber ->> Subscriptions Container: Request subscription (with access token)
-  Subscriptions Container ->> Authorization Server: Verify authorization
-  Authorization Server ->> Subscriptions Container: capabilities
-  Subscriptions Container ->> Subscriber: Subscription response (with webid & unsubscribe)
   loop for each notification
```
[the remark this answers is not in the document]
Comment (acoburn) [reply notifications-flow-target.mmd:6]: No strong opinion on this. I suspect that the loop construct could be omitted

[the remark this answers is not in the document]
Comment (elf-pavlik) [reply notifications-flow-target.mmd:6]: Each notification results in separate request to and response from the target, in contrast to connecting to the source where all notifications are streamed in the same ongoing response.

If you look at https://github.com/solid/notifications/pull/108#issuecomment-1272336602 subscription flow  has a block where notifications flow would conceptually fit in. We still have open question of adding unsubscribe and once (if) we add it, having single request+response to the target between subscribe and unsubscribe doesn't illustrate how it works very well.

In short, the diagrams are non-normative and for the sake of illustrating how things work, the loop IMO comes helpful.

**Comment by elf-pavlik** [comment]
>  As I understand it, the "publisher" is a role of creating/producing/sending/delivering/distributing a notification, and that the publication/having a representation of the topic resource is done by some resource server.

How publisher is deploying there services is out of scope. Still both the resource which is referenced as the *topic* of the subscription and the party which is delivering notifications to the target, or providing the notifications source, are working together. Since we don't define any details how the resource server and the notifying party are working together, they could be part of the same application or separate services using implementation specific way of communicating.
Actually the subscription manager could also be a separate service which coordinates with other two mentioned roles. This way we have 3 distinct responsibilities under a single umbrella of the *publisher*:

* resource server hosting the *topic*
* subscription manager - we define how subscriber can discover it
* notifying party - we only define how subscriber either fetches notifications for source or exposes target to which notifications get delivered

This conversation should leave this PR BTW so that we can close it.

**Comment by csarven** [comment]
The context of the interaction here is specifically about a sender delivering a notification to a target. I have no objection to using general terms when necessary, and not in place of more specific terms. I want to know if you have an objection to using sender so we can close the PR.

**Review by csarven — APPROVED** [review]

File: notifications-flow-target.mmd (hunk 6 above)
[the remark this answers is not in the document]
Comment (csarven) [reply notifications-flow-target.mmd:6]: Perhaps +0.5 for simplified (dropping loop) but no objection to leave loop in.

**Comment by csarven** [comment]
Shouldn't the resource that advertises the discovery (notification channel) resource be included in the diagram in https://solid.github.io/notifications/protocol#solid-notifications-flow-protocol ?
